src/infrastructure/repositories/associated-figma-design-repository.ts

```diff
@@ -22,53 +21,75 @@ export class AssociatedFigmaDesignRepository {
 	): Promise<AssociatedFigmaDesign> => {
 		const dbModel = this.mapToDbModel(associatedFigmaDesign);
 
-		const result = await prismaClient.get().associatedFigmaDesign.upsert({
+		const record = await prismaClient.get().associatedFigmaDesign.upsert({
 			create: dbModel,
 			update: dbModel,
 			where: {
-				fileKey_nodeId_connectInstallationId: {
+				fileKey_nodeId_associatedWithAri_connectInstallationId: {
 					fileKey: dbModel.fileKey,
 					nodeId: dbModel.nodeId,
+					associatedWithAri: dbModel.associatedWithAri,
 					connectInstallationId: dbModel.connectInstallationId,
 				},
 			},
 		});
-		return this.mapToDomainModel(result);
+		return this.mapToDomainModel(record);
 	};
 
 	findManyByFileKeyAndConnectInstallationId = async (
 		fileKey: string,
 		connectInstallationId: number,
 	): Promise<AssociatedFigmaDesign[]> => {
-		const result = await prismaClient.get().associatedFigmaDesign.findMany({
+		const records = await prismaClient.get().associatedFigmaDesign.findMany({
 			where: { fileKey, connectInstallationId },
 		});
 
-		return result.map(this.mapToDomainModel);
+		return records.map(this.mapToDomainModel);
 	};
 
-	deleteByDesignIdAndConnectInstallationId = async (
+	findByDesignIdAndAssociatedWithAriAndConnectInstallationId = async (
 		designId: FigmaDesignIdentifier,
+		associatedWithAri: string,
 		connectInstallationId: number,
-	): Promise<AssociatedFigmaDesign> => {
+	): Promise<AssociatedFigmaDesign | null> => {
+		const record = await prismaClient.get().associatedFigmaDesign.findFirst({
+			where: {
+				fileKey: designId.fileKey,
+				nodeId: designId.nodeId ?? '',
+				associatedWithAri: associatedWithAri,
+				connectInstallationId,
+			},
+		});
+
+		if (record === null) return null;
+
+		return this.mapToDomainModel(record);
+	};
+
+	deleteByDesignIdAndAssociatedWithAriAndConnectInstallationId = async (
+		designId: FigmaDesignIdentifier,
+		associatedWithAri: string,
+		connectInstallationId: number,
+	): Promise<AssociatedFigmaDesign | undefined> => {
 		try {
-			const result = await prismaClient.get().associatedFigmaDesign.delete({
+			const record = await prismaClient.get().associatedFigmaDesign.delete({
 				where: {
-					fileKey_nodeId_connectInstallationId: {
+					fileKey_nodeId_associatedWithAri_connectInstallationId: {
 						fileKey: designId.fileKey,
 						nodeId: designId.nodeId ?? '',
+						associatedWithAri: associatedWithAri,
 						connectInstallationId,
 					},
 				},
 			});
 
-			return this.mapToDomainModel(result);
+			return this.mapToDomainModel(record);
 		} catch (e: unknown) {
 			if (
 				e instanceof PrismaClientKnownRequestError &&
 				e.code === PrismaErrorCode.RecordNotFound
 			) {
-				throw new RepositoryRecordNotFoundError(e.message);
```

Comment:
/nit Not sure about this `deleteXXX` method behaving differently to others, but I get we don't really have a naming convention for delete methods that don't throw on missing records. The return type does kind of indicate this, so this is probably okay.

Reply:
That's a good question, Justin!

What do you think about using the following convention?

- `deleteXxx` returns a deleted record or `null` if it does not exist. Likely, this is what we need in the majority of cases due to the idempotency requirement and concurrency -- usually, we want to make sure that a record is deleted but not necessarily delete it with a particular method call. If we look at other team's projects, all our `deleteXxx` methods provide this contract and use this convention.
- `deleleXxxOrThrowIfMissing returns a deleted record or throw if a record is missing. This is less likely but still possible contract for our methods.

Please, let me know, Justin, whether you have any concerns or find these names confusing or clunky.

Reply:
Added integration tests for `AssociatedFigmaDesignRepository.deleteByDesignIdAndAssociatedWithAriAndConnectInstallationId` to cover its tricky error handling.